### PR TITLE
test(resourceDirectory): add check for directory/file call order

### LINKED ISSUE #400: [Tests] Add check for directory and file level operation ordering
For the renaming of resource categories and collections/subcollections, we perform both directory and file level operations. For directory level operations, we use git trees, but since we use a commit hash from the start of the operation, any directory level operations need to be done before file level operations. We should add this into our tests for renaming folders in ResourceDirectoryService and CollectionDirectoryService - we could look into [jest-extended for .toHaveBeenCalledBefore](https://github.com/jest-community/jest-extended#tohavebeencalledbefore).

### PR DESCRIPTION
**Pull request by seaerchin**
## Problem
See #400 for a brief description of the issue.

Closes #400

## Solution
1. add `jest-extended`
2. check that directory level calls occur before file level reads

**New dev dependencies**:

- `jest-extended` : package to extend `jest` and add more matchers

### CODE REVIEW

File: services/directoryServices/__tests__/ResourceDirectoryService.spec.js [hunk 1]
```diff
@@ -284,21 +287,26 @@ describe("Resource Directory Service", () => {
         fileName: INDEX_FILE_NAME,
         directoryName: `${resourceRoomName}/${newDirectoryName}`,
       })
+      expect(getResourceSpy).toHaveBeenCalledBefore(mockGitHubService.update)
```
Comment (alexanderleegs) [inline services/directoryServices/__tests__/ResourceDirectoryService.spec.js:290]: Not sure how useful it is that the util function is called before the update - I think we might want to check that the `rename` operation from `baseDirectoryService` is called before `update` operation from `GitHubService`

File: services/directoryServices/__tests__/ResourceDirectoryService.spec.js [hunk 2]
```diff
@@ -322,6 +330,7 @@ describe("Resource Directory Service", () => {
         fileName: INDEX_FILE_NAME,
         directoryName: `${resourceRoomName}/${slugifiedResourceCategory}`,
       })
+      expect(getResourceSpy).toHaveBeenCalledBefore(mockGitHubService.update)
```
Comment (alexanderleegs) [inline services/directoryServices/__tests__/ResourceDirectoryService.spec.js:333]: see above!

**Review by alexanderleegs — CHANGES_REQUESTED** [review]
left comments!

File: services/directoryServices/__tests__/ResourceDirectoryService.spec.js (hunk 1 above)
Comment (seaerchin) [reply services/directoryServices/__tests__/ResourceDirectoryService.spec.js]: thanks for the catch! updated accordingly

**Review by alexanderleegs — APPROVED** [review]
lgtm!

**Pull request merged**